src/test/compile-fail/default_ty_param_conflict.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fmt::Debug;
+
+// Example from the RFC
+fn foo<F:Default=usize>() -> F { F::default() }
+fn bar<B:Debug=isize>(b: B) { println!("{:?}", b); }
+
+fn main() {
+    // Here, F is instantiated with $0=uint
+    let x = foo();
+
+    // Here, B is instantiated with $1=uint, and constraint $0 <: $1 is added.
+    bar(x);
+}
```

Comment:
I was working on improving it this afternoon/evening but got side tracked by trying to get type aliases to work correctly. Going to try to finish it tomorrow morning PST.